apps/frontend/src/components/Decorators/Contexable.tsx

```diff
@@ -0,0 +1,33 @@
+import { useContextMenu } from '@/hooks/useContextMenu';
+import { ContextMenuAction } from '@/types/contextMenu';
+import { MouseEvent } from 'react';
+import ContextMenu from '../ContextMenu/ContextMenu';
+
+// TODO Renames the component
+interface ContextMenuProps {
+    children: React.ReactElement;
+    actions: ContextMenuAction[];
+    className: string;
+}
+const Contextable: React.FunctionComponent<ContextMenuProps> = (props) => {
+    // destructure our state and set state functions from our custom hook
+    const { clicked, setClicked, coords, setCoords } = useContextMenu();
```

Comment:
The returned element of the hook should be a tuple, in order to keep a uniform code style with native React hooks.
```typescript
const [ clicked, setClicked, coords, setCoords ] = useContextMenu();
```

Reply:
If I do this its mean that useContextMenu() return an array, and that's is not the case here.

I think its should return an object that includes the state variables clicked and coords, as well as the corresponding setter functions because this makes it easier for consuming components to access and update the context menu state.